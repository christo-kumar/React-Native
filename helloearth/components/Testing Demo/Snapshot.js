/* eslint-disable react/self-closing-comp */
/* eslint-disable no-unused-vars */
/* eslint-disable prettier/prettier */

import React from 'react';
import { View, StyleSheet, Text } from 'react-native';

const styles = StyleSheet.create({
    body: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    }
});

export const Snapshot = () => {
    return (
        <View>
            <Text> Hello World </Text>
        </View>
    );
}

