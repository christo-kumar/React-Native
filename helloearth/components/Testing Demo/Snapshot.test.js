/* eslint-disable react/self-closing-comp */
/* eslint-disable no-unused-vars */
/* eslint-disable prettier/prettier */

import React from 'react';
import { FlatList, Text } from 'react-native';
import renderer from 'react-test-renderer';
import { Snapshot } from './Snapshot';

//Component name can be anything
//Component snapshot would be created in Snapshot.test.js.snap
//If UI changes later this test would fail
//Delete the snapshot if UI changes later
test('render conrrectly', () => {
    const tree = renderer.create(<Snapshot />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('test rendering of flatList', () => {
    const tree = renderer.create(
        <FlatList
            data={['1', '2', '3']}
            keyExtractor={(index, value) => index.toString()}
            renderItem={({ item }) => <Text> item </Text>}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
