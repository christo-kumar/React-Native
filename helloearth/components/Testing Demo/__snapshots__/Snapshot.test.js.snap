// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render conrrectly 1`] = `
<View>
  <Text>
     Hello World 
  </Text>
</View>
`;

exports[`test rendering of flatList 1`] = `
<RCTScrollView
  data={
    Array [
      "1",
      "2",
      "3",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
         item 
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
         item 
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
         item 
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
